### PR TITLE
New version of Features

### PR DESCRIPTION
**Pull request by MarinaKalashina**
In this commit:
- page structure reworked, TOC added
- new texts, images, and gifs

### CODE REVIEW

File: features.html [hunk 1]
```diff
@@ -1,14 +1,358 @@
...
+<p></p>
+<p>The highlighting palette is fully customizable for both Darcula and Light color schemes.</p>
+
+
+
+  <h3>Quick docs and Quick definition</h3>
+<table border="1" cellpadding="5">
+  <tr>
+    <td width="50%" >At any code element, including library functions, invoke <strong>Quick Doc</strong> (<code>Ctrl+Q / F1</code>) to view the documentation popup. If there is a piece of code in the docs, it will be formatted and highlighted properly:</td>
+    <td>To view the implementation or declaration of an element without leaving your current context, call <strong>Quick Definition</strong> by pressing <code>Ctrl+Shift+I / ⌥Space</code>:</td>
```
[the remark this answers is not in the document]
Comment (Undin) [reply features.html]: Looks like we need another font for shortcuts

File: features.html [hunk 2]
```diff
@@ -1,14 +1,358 @@
...
+  td {
+    padding: 5px; /* Поля в ячейках */
+    vertical-align: top; /* Выравнивание по верхнему краю ячеек */
+  }
+</style>
+
+{% include features-intro.html %}
+
+
+<h2 id="editor" align="center">Smart editor</h2>
```
Comment (Undin) [inline features.html]: Do we really need `align="center"`?

File: features.html [hunk 3]
```diff
@@ -1,14 +1,358 @@
...
+<p>Hints for function argument names, types, and lambda parameters are displayed in the editor, and IntelliJ Rust adjusts them as you type. If you prefer not to see some of the hints or hide them entirely, you can always tune that in the settings.</p>
+
+
+<table border="1" cellpadding="5">
+  <tr>
+    <td width="50%" >If you press <code>Ctrl</code> and hover the mouse over a hint, it will turn into an interactive inlay:</td>
+    <td>You can also get the type information by calling <strong>Type Info</strong> (<code>Ctrl + Shift + P / ⌃⇧P</code>) for a selected expression:</td>
+  </tr>
+  <tr>
+    <td><img class="gif" data-wait="true" data-scope="gif-scope" width="500"
```
Comment (Undin) [inline features.html]: Maybe we can create "include" similar with `_includes/feature-gfy.html` to simplify gif insertions?

Comment (Undin) [reply features.html]: I've already implemented such include html in master.
See at [gif-img.html](https://github.com/intellij-rust/intellij-rust.github.io/blob/master/_includes/gif-img.html). Note, it should be slightly modified to use absolute width of inserting picture

File: features.html [hunk 4]
```diff
@@ -1,322 +1,475 @@
...
+  <div class="item_col2">
+    <img src="/assets/features/code_generation/add_derive_clause_2.png" alt="add derive clause intention"/>
+    <p></p>
+    <p>You can quickly add <code>#[derive]</code> clauses and <code>impl</code> blocks for structs and enums, and perform struct-to-tuple and tuple-to-struct conversion</p>
   </div>
-
-  <div class="feature">
-    <h3>File Structure</h3>
-    <p><kbd>Ctrl+F12</kbd>, <kbd>Alt+7</kbd> brings up an overview of the current file.</p>
+  <div class="item_bottomtext">
```
Comment (Undin) [inline features.html]: This item looks not uniform with others
image

File: features.html [hunk 5]
```diff
@@ -1,322 +1,475 @@
...
-    <h3>Goto Symbol</h3>
-    <p><kbd>Ctrl+Shift+Alt+N</kbd> searches for any symbol (types, methods, functions, fields) by name. Together with
-      <kbd>Ctrl+N</kbd> it is the main method to explore large projects.</p>
+<h2 id="intentions" align="left">Intentions and quick-fixes</h2>
+<div class="feature-section">
+  <div class="item_col1">
+    <!--img class="gif" data-wait="true" data-scope="gif-scope" style="width: auto"
+         src="/assets/features/code_generation/missing_fields.png" data-gif="/assets/features/code_generation/missing_fields.gif"
+         alt="missing fields intention"/-->
+    <img src="/assets/features/code_generation/missing_fields_img.png" alt="missing fields intention"/>
```
Comment (Undin) [inline features.html]: Why png instead of gif?

File: features.html [hunk 6]
```diff
@@ -1,322 +1,475 @@
...
+    <p></p>
+    <p>The <i>Add missing fields</i> quick fix helps you interactively define field values of struct literals.</p>
   </div>
-
-  <div class="feature">
-    <h3>Goto Super</h3>
-    <p><kbd>Ctrl+U</kbd> navigates to the parent module.</p>
+  <div class="item_col2">
+    <img src="/assets/features/code_generation/add_derive_clause_2.png" alt="add derive clause intention"/>
```
Comment (Undin) [inline features.html]: Same about gif vs png

File: features.html [hunk 7]
```diff
@@ -1,322 +1,475 @@
...
-      </div>
-
-      <div class="feature">
-        {% include feature-gfy.html n="GlisteningCostlyAntarcticfurseal" %}
-        <p><b>Implement methods</b>: quickly scafold an impl.</p>
-      </div>
-    </div>
-
+  <div class="item_bottomtext3">
+    <p>At any place in your code, call <strong>Rector This</strong> (<code>Ctrl+Shift+Alt+T / ⌃T</code>) to view the list of all context-depend refactorings. As an example, here is how you can easily put the result of an expression into a variable, affecting all usages:
```
Comment (Undin) [inline features.html]:
```suggestion
    <p>At any place in your code, call <strong>Refactor This</strong> (<code>Ctrl+Shift+Alt+T / ⌃T</code>) to view the list of all context-depend refactorings. As an example, here is how you can easily put the result of an expression into a variable, affecting all usages:
```

Comment (Undin) [inline features.html]: Maybe move part about `Refactor This` action into description and convert this block into `Extract Variable` section?

File: features.html [hunk 8]
```diff
@@ -1,322 +1,475 @@
...
-      <div class="feature">
-        <p><b>Expand module</b>: invoke this refactoring inside <code>foo.rs</code> file to get
-          <code>foo/mod.rs</code>. Also available as a quick fix for unresolved module declaration.</p>
-      </div>
-
-      <div class="feature">
-        <p><b>Introduce variable</b>: use <kbd>Ctrl+Alt+V</kbd> or <code>.let</code> postfix template to extract a variable from an expression.</p>
-      </div>
-    </div>
+  <div class="item_bottomtext2">
```
Comment (Undin) [inline features.html]: Maybe it's better to convert this section into two-columns one?

File: features.html [hunk 9]
```diff
@@ -1,322 +1,475 @@
...
 
-<div class="feature-group">
-  <h2>Reading code</h2>
-  {% include feature-pic.html i="feature-reading" %}
-
-  <div class="feature half">
-    <h3>Quick documentation</h3>
-    <p><kbd>Ctrl+Q</kbd> shows docs for the symbol at caret.</p>
+<h2 id="analysis" align="left">Code analysis</h2>
+<div class="feature-section-3cols">
```
Comment (Undin) [inline features.html]: Let's do not use three columns sections because in this case a column is too small

File: features.html [hunk 10]
```diff
@@ -1,322 +1,475 @@
...
+  <div class="item_toptext_3cols">
+    <p>IntelliJ Rust uses a set of inspections to monitor code as you type. You can configure the list of them, and set the severities and scopes in <strong>Settings / Preference | Editor | Inspections</strong>. By default, most inspections work on-the-fly and suggest quick-fixes to be applied right away. Find a few examples below:
+    </p>
   </div>
-
-  <div class="feature half">
-    <h3>Documentation highlighting</h3>
-    <p>IntelliJ Rust highlights important parts of documentation comments, so they never look so bland.</p>
+  <div class="item_col1_3">
+    <p>Type check errors</p>
```
Comment (Undin) [inline features.html]: We have different layout for item in column in two-columns sections. Let's unify them

File: features.html [hunk 11]
```diff
@@ -1,322 +1,475 @@
...
-    Alternatively, you can use the icon in the gutter area.
+<h2 id="testing" align="left">Testing</h2>
+<div class="feature-section">
+  <div class="item_toptext">
+    <p>Gutter icon menu is the quickest way to start a run or debug (in CLion) session for your tests:</p>
+    <img src="/assets/features/testing/test_gutter.png" width="470" align="center" alt="run tests from gutter menu"/>
   </div>
+  <div class="item_bottomtext">
+    <p>When you run tests using gutter icons or by calling <code>cargo test</code>, IntelliJ Rust employs the IDE's built-in test runner to help you explore the results. It shows the status and duration of each test, and you can sort, save, or import/export the results, as well as jump to the corresponding code:</p>
+    <img src="/assets/features/testing/test_runner.png" width="470" align="center" alt="test runner"/>
```
Comment (Undin) [inline features.html]: It would be great to have screenshot with failed tests

**Comment by MarinaKalashina** [comment]
Closing this PR due to postponing the rework of 'Features'. I'll extract the new chapter in QSG and the fixed year in the footer into separate requests.

**Pull request closed without merging**